src/Chapter2.hs

```diff
@@ -727,7 +739,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate (x:xs) = concatMap (replicate x) (smartReplicate xs)
```

Comment:
I think this implementation is not correct. It will replicate every number in the list (except the first one) as many times as the value of the head of the list. But what you want is to replicate each number as many times as the value of that number itself.

Try to implement this function by `concatMap myReplicate` and define `myReplicate` as a helper function that does the main work. After that, you probably can replace your custom `myReplicate` with a lambda function for shorter code 🙂